changelog.md

```diff
@@ -1,3 +1,68 @@
+## Code Dx Jenkins Plugin 4.0.0 `Released 5/10/2023`
+
+### Changes
+- The "API Key" field now uses a `credential` instead of a `string` value
+- The "API Key" field ID (`key`) has been renamed to `keyCredentialId`
+- When configuring Freestyle projects, "Configure" permission on the project is required to make changes
+- When using the Pipeline Syntax Generator tool, the global "Administer" permission is required to validate the form and list projects
+  - It can still be used without these permissions, but the "Code Dx Project" field will need to be filled manually since the drop-down will remain empty
+
+### Notice
+
+This release was performed in cooperation with the Jenkins team and is a response to several reported vulnerabilities:
+
+- [CSRF vulnerability and missing permission checks in Code Dx Plugin](https://www.jenkins.io/security/advisory/2023-05-16/#SECURITY-3118)
+- [Missing permission checks in Code Dx Plugin](https://www.jenkins.io/security/advisory/2023-05-16/#SECURITY-3145)
+- [API keys stored and displayed in plain text by Code Dx Plugin](https://www.jenkins.io/security/advisory/2023-05-16/#SECURITY-3146)
+
+It is a breaking change and will require reconfiguring Pipeling and Freestyle Projects.
```

Comment:
Should this be Pipeline instead of Pipeling?

Reply:
Ah yep typo, ty